packages/@aws-cdk/aws-fsx/README.md

```diff
@@ -17,8 +17,188 @@
 ---
 <!--END STABILITY BANNER-->
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+[Amazon FSx](https://docs.aws.amazon.com/fsx/?id=docs_gateway) provides fully managed third-party file systems with the
+native compatibility and feature sets for workloads such as Microsoft Windows–based storage, high-performance computing,
+machine learning, and electronic design automation.
+
+Amazon FSx supports two file system types: [Lustre](https://docs.aws.amazon.com/fsx/latest/LustreGuide/index.html) and
+[Windows](https://docs.aws.amazon.com/fsx/latest/WindowsGuide/index.html) File Server.
+
+## FSx for Lustre
+
+Amazon FSx for Lustre makes it easy and cost-effective to launch and run the popular, high-performance Lustre file
+system. You use Lustre for workloads where speed matters, such as machine learning, high performance computing (HPC),
+video processing, and financial modeling.
+
+The open-source Lustre file system is designed for applications that require fast storage—where you want your storage
+to keep up with your compute. Lustre was built to solve the problem of quickly and cheaply processing the world's
+ever-growing datasets. It's a widely used file system designed for the fastest computers in the world. It provides
+submillisecond latencies, up to hundreds of GBps of throughput, and up to millions of IOPS. For more information on
+Lustre, see the [Lustre website](http://lustre.org/).
+
+As a fully managed service, Amazon FSx makes it easier for you to use Lustre for workloads where storage speed matters.
+Amazon FSx for Lustre eliminates the traditional complexity of setting up and managing Lustre file systems, enabling
+you to spin up and run a battle-tested high-performance file system in minutes. It also provides multiple deployment
+options so you can optimize cost for your needs.
+
+Amazon FSx for Lustre is POSIX-compliant, so you can use your current Linux-based applications without having to make
+any changes. Amazon FSx for Lustre provides a native file system interface and works as any file system does with your
+Linux operating system. It also provides read-after-write consistency and supports file locking.
+
+### Installation
+
+Import to your project:
 
 ```ts
 import fsx = require('@aws-cdk/aws-fsx');
 ```
+
+### Basic Usage
+
+Setup required properties and create:
+
+```ts
+const stack = new Stack(app, 'Stack');
+const vpc = new Vpc(stack, 'VPC');
+const storageCapacity = 1200;
+const lustreConfiguration = {
+  deploymentType: FsxLustreDeploymentType.SCRATCH_2
+};
+
+const fileSystem = new FsxLustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration,
+  storageCapacity,
```

Comment:
A unitless number scares me. Can we make this `storageCapacityGiB`, `storageCapacityGB`, `storageCapacityTiB`, or whatever the case may be?

Reply:
I've inlined this along with the lustreConfiguration so there is no variable and line is `storageCapacity: 1200`. Would you still rather have a variable named `storageCapacityGiB`?

Did you mean rename the attribute itself to `storageCapacityGiB`? I could do that but it would be inconsistent with how CloudFormation names it.

Reply:
Yes, I meant rename the property. I'm okay taking the hit being not *exactly* the same. The prefix is the same (so people typing will autocomplete to the prop they expect) and I feel the additional clarity is worth it.

Not all AWS services are consistent with one another, but we can at least be consistent within the CDK.

Reply:
Fixed.